scripts/SREP/mg-log-extractor/script.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# exit on error
+set -e
+
+export AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY
+
+hiveconfig=$(oc get hiveconfig hive -o json)
+if [ -z "$hiveconfig" ] ; then
+  echo "Error: Unable to retrieve hiveconfig."
+  exit 10
+fi
+
+bucket=$(echo "$hiveconfig" | jq -r '.spec.failedProvisionConfig.aws.bucket')
+if [ -z "$bucket" -o "$bucket" == "null" ] ; then
+  echo "Error: Unable to determine S3 bucket from hiveconfig."
+  exit 10
+fi
+
+region=$(echo "$hiveconfig" | jq -r '.spec.failedProvisionConfig.aws.region')
+if [ -z "$region" -o "$region" == "null" ] ; then
+  echo "Error: Unable to determine AWS region from hiveconfig."
+  exit 10
+fi
+
+secret_name=$(echo "$hiveconfig" | jq -r '.spec.failedProvisionConfig.aws.credentialsSecretRef.name')
+if [ -z "$secret_name" -o "$secret_name" == "null" ] ; then
+  echo "Error: Unable to determine AWS credentials secret name from hiveconfig."
+  exit 10
+fi
+
+credentials_secret=$(oc get secret -n hive -o json "$secret_name")
+if [ -z "$credentials_secret" -o "$credentials_secret" == "null" ] ; then
+  echo "Error: Unable to retrieve credentials secret [$secret_name]."
+  exit 10
+fi
+
+AWS_ACCESS_KEY_ID=$(echo "$credentials_secret" | jq -r '.data.aws_access_key_id' | base64 -d)
+if [ -z "$AWS_ACCESS_KEY_ID" -o "$AWS_ACCESS_KEY_ID" == "null" ] || [[ "$AWS_ACCESS_KEY_ID" != AKIA* ]] ; then
+  echo "Error: Unable to retrieve aws_access_key_id from secret [$secret_name] in the hive namespace."
+  exit 10
+fi
+
+AWS_SECRET_ACCESS_KEY=$(echo "$credentials_secret" | jq -r '.data.aws_secret_access_key' | base64 -d)
+if [ -z "$AWS_SECRET_ACCESS_KEY" -o "$AWS_SECRET_ACCESS_KEY" == "null" ]; then
+  echo "Error: Unable to retrieve aws_secret_access_key from secret [$secret_name] in the hive namespace."
+  exit 10
+fi
+
+s3_log_dir="$TARGET_LOG_DIR"
+if [ -z "s3_log_dir" ]; then
+  echo "Error: s3_log_dir unspecified."
+  exit 10
+fi
+
+aws s3 sync "s3://${bucket}/${s3_log_dir}" . > /dev/null 2>&1
+
+for logbundle in ./*.tar.gz; do
+  cat "$logbundle" > out.tar.gz
+done
+
+tar zcf - out.tar.gz
```

Comment:
Does this works as expected?

Reply:
I copied from here https://github.com/openshift/managed-scripts/blob/main/scripts/CEE/get-kafka-instance-state/get-kafka-instance-state.sh#L49
as the OSD card said to. To be honest I haven't gotten this to work. Running `ocm backplane testjob logs <job>` returns a bunch of data that looks like a tar.gz but when trying to unzip it, I get errors. I've asked multiple times in the backplane slack channel but haven't had help.

Reply:
@sedroche is looking at this.

Reply:
@fahlmant I did some testing (on a simple test not your script) but pipe-ing the logs works by getting the logs from the pod directly using oc logs. The pod will be the job name in openshift-backplane-managed-scripts namespace. Can you test your script using those logs?
I think we will need some chnages in backplane to fix this

Reply:
Let's make this in as we have a workaround to use "oc logs"